e2e-polybft/e2e/bridge_test.go

```diff
@@ -287,14 +285,164 @@ func TestE2E_Bridge_Transfers(t *testing.T) {
 
 		// the transactions are mined and state syncs should be executed by the relayer
 		// and there should be a success events
-		var stateSyncedResult contractsapi.StateSyncResultEvent
-
 		logs, err := getFilteredLogs(stateSyncedResult.Sig(), initialBlockNum, finalBlockNum, childEthEndpoint)
 		require.NoError(t, err)
 
 		// assert that all state syncs are executed successfully
 		checkStateSyncResultLogs(t, logs, transfersCount)
 	})
+
+	t.Run("non native ERC20 deposit and withdraw", func(t *testing.T) {
+		rootchainTxRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(cluster.Bridge.JSONRPCAddr()))
+		require.NoError(t, err)
+
+		childchainTxRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithClient(validatorSrv.JSONRPC()))
+		require.NoError(t, err)
+
+		// Deploy ERC20 contract to the rootchain
+		rootchainDeployer, err := rootHelper.DecodePrivateKey("")
+		require.NoError(t, err)
+
+		txn := &ethgo.Transaction{To: nil, Input: contractsapi.RootERC20.Bytecode}
+		receipt, err := rootchainTxRelayer.SendTransaction(txn, rootchainDeployer)
+		require.NoError(t, err)
+		require.NotNil(t, receipt)
+
+		contractAddress := receipt.ContractAddress
```

Comment:
I'd suggest renaming it to rootTokenAddr (just for better clarity):
```suggestion
		rootTokenAddr := receipt.ContractAddress
```

Reply:
Done